### PR TITLE
Basic cardinity fiat payments

### PR DESCRIPTION
**Pull request by mdomasevicius**
image
image
image
image
image
image

### CODE REVIEW

File: src/Pages/Onboarding/steps/TopUpModal/TopUpModal.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+  const isFree = () => registrationOption === 'free'
+  const isRegistrationFeeReceived = isFree()
+    ? true
+    : money.flooredAmount(identity.balance, 3) >= registrationInfo.flooredFee
+  const registrationOptions: Option[] = useMemo(
+    () => [
+      { value: 'free', label: 'Register for free', disabled: !isFreeRegistrationEligible },
+      { value: 'myst', label: 'Deposit MYST token' },
+      { value: 'fiat', label: 'Buy MYST with Credit card' },
```
Comment (tomasmik) [inline src/Pages/Onboarding/steps/TopUpModal/TopUpModal.tsx:63]: This is gonna be nitpicking but you can also use a debit card 🤷🏻

File: src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+  return (
+    <a download={name} href={url} style={{ display: isLoading ? 'none' : undefined }}>
+      Download Invoice
+    </a>
+  )
+}
+
+const PriceInfo = ({ rates, mystAmounts }: { rates: Money[]; mystAmounts: { [key: string]: number } }) => {
+  const amounts = rates.map((r) => `1 ${r.currency} (${mystAmounts[r.currency]} MYST)`).join(' / ')
```
Comment (tomasmik) [inline src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx:273]: Well... not exactly. Minimum amount is 1USD, so right now it'd be 0.88 Euro etc.

File: src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        },
+      })
+
+      const html = validateOrderAndReturnSecureForm(order)
+      const data = buildFormData(html)
+
+      if (!data) {
+        throw new Error('Cardinity secure form not provided')
+      }
+      await api.payment.invoice(identity.id, order.id)
```
Comment (Guillembonet) [inline src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx:150]: should this be here? I see it is also called in DownloadInvoice

File: src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        </button>
+        <input type="hidden" name="amount" value={state.formData['amount']} />
+        <input type="hidden" name="country" value={state.formData['country']} />
+        <input type="hidden" name="currency" value={state.formData['currency']} />
+        <input type="hidden" name="order_id" value={state.formData['order_id']} />
+        <input type="hidden" name="project_id" value={state.formData['project_id']} />
+        <input
+          type="hidden"
+          name="return_url"
+          value="https://pilvytis-sandbox.mysterium.network/api/v2/payment/cardinity/redirect"
```
Comment (Guillembonet) [inline src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx:229]: @tomasmik is this right? Maybe this url should come from node somehow and depend on pilvytis address set there?

File: src/commons/country.ts [hunk 5]
```diff
@@ -14,3 +14,243 @@ export const countryName = (countryCode: string): string => {
   if (countryCode !== '') return regionNames.of(countryCode)
   else return 'Unknown'
 }
+
+export const countryNames: { [key: string]: string } = Object.freeze({
```
Comment (Guillembonet) [inline src/commons/country.ts:18]: isn't it doing the same as `countryName`?

**Review by Guillembonet — COMMENTED** [review]
That was a long one (don't reply with: that's what she said)

File: src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx (hunk 4 above)
Comment (tomasmik) [reply src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx:229]: Indeed this should be a non sandbox url or better yet a one from node itself.

File: src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx (hunk 2 above)
Comment (mdomasevicius) [reply src/Pages/Onboarding/steps/TopUpModal/Fiat/Fiat.tsx:273]: The 1 USD/EUR/GPB does not change, the `${mystAmounts[r.currency]}` si different

File: src/commons/country.ts (hunk 5 above)
Comment (mdomasevicius) [reply src/commons/country.ts:18]: expand

Comment (mdomasevicius) [reply src/commons/country.ts:18]: `countryName` was designed for translating code to name, that said `countryNames` is a set of options for drop down.

**Review by tomasmik — APPROVED** [review]

**Review by Guillembonet — APPROVED** [review]

**Pull request merged**